Moved Inference Implementation and Testing Scripts from Testbed

Riverflame-dev opened the pull request:
models/
- All 16 LD models and groundtruth
- 1 RC model (16 classes)

flight/vision/ld/
- landmark_detector.py

flight/vision/rc/
- region_classifier.py

tests/
- test_LD_inference.py: runs LD inference with given test data
- test_RC_inference.py: runs RC inference with given test data
- test_rc_ld_inference.py: runs an RC inference with given test data and uses the RC net output to activate appropriate LD inferences for the images classified into the region(s). Evaluation result saved under tests/inference_output

tests/data/
- full_inference/: 5 images + labels from each region's test set

Ibrassow commented:
- Move the contents of requirements.txt in the tests folder to the root one
- Missing ld and rc inference code in vision
- Refrain from using sys.path.append('..’) and use absolute imports (from the root dir) for modules, e.g. from flight.vision.rc.XXX 
- test_RC_inference.py, line 41: I would set the region IDs of interest in a configuration file placed in the configuration folder. Could have also a separate region configuration for the actual flight, tests, or other specific tests.

Riverflame-dev commented:
@Ibrassow Fixed issues mentioned in the review:
- Updated requirements.txt in the root one and removed from tests/
- Added ld and rc inference code to the commit
- Removed sys.path.append('..’) and absolute imports in RC, replaced with environment variable for project root path 
- Created inference configuration file for region configuration (and model path etc for future updates)

Updates and File Structure:

configuration/
- inference_config.yml: created inference configuration file for region configuration, future update will include model name, path, output path etc

flight/vision/ld/
- landmark_detector.py: updated detection method to return only centroid_xy, centroid_latlons, landmark_class
flight/vision/rc/
- region_classifier.py: created method to load region configuration from inference_config.yml, dynamic construction of model path 

flight/vision/
- ml_pipeline.py: defined class and methods to processes a series of frames from camera feeds, performs region classification and detects landmarks within those regions

tests/
- fake_camera_generator.py: simulating camera feeds paired with camera id
- test_ml_pipeline.py: runs the ml_pipeline class method with fake camera feeds

Ibrassow reviewed (approved):
Good job
- Corrected the path throughout the code in the last commits
- The camera must be mocked in the next iteration
- No inference data file?
- Inference time is interestingly slow
